tiledb/sm/query/writer.cc

```diff
@@ -2782,15 +2795,16 @@ Status Writer::prepare_tiles(
   auto buffer = (uint64_t*)it->second.buffer_;
   auto buffer_var = (uint64_t*)it->second.buffer_var_;
   auto buffer_validity = (uint64_t*)it->second.validity_vector_.buffer();
-  auto buffer_size = it->second.buffer_size_;
+  auto buffer_size =
+      get_offset_buffer_size(attribute, *it->second.buffer_size_);
```

Comment:
I think the `get_offset_buffer_size` function name is misleading because the "offset buffer" wording only applies to var-size attributes. I think the right thing to do the following:
```
auto buffer_size ==
  var_size ? get_offset_buffer_size(*it->second.buffer_size_) : *it->second.buffer_size_;
```

Modify the `get_offset_buffer_size` to assume it is only being used on a var-sized attribute.

L2563 is OK as it is because that path only applies to var-size attributes.

Reply:
Agreed. it's clearer like that